File: 01_Array_Hashings/07_Product_Of_Array_Except_Self/238_Product_Of_Array_Except_Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];

        for (int i=0;i<nums.length; i++){
            System.out.print(nums[i]);

            int p=1;
            for (int p_i=0; p_i<i; p_i++){
                p *= nums[p_i];                
            }

            int s=1;
            for (int s_i=nums.length-1; s_i>i; s_i--){
                s *= nums[s_i];
            }
            res[i]= p*s;
        }

        return res;
    }
}  

File: 01_Array_Hashings/07_Product_Of_Array_Except_Self/238_Product_Of_Array_Except_Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = []
        for i in range(len(nums)):
            pref = 1
            for i_pref in range(0, i):
                pref *= nums[i_pref]
            suff = 1
            for i_suff in range(len(nums)-1, i, -1):
                suff *= nums[i_suff]

            res.append(pref * suff)
                

        
        return res
        